charms/benchmark/__init__.py

```diff
@@ -118,9 +120,26 @@ def start():
         If the cabs-collector charm is installed, take a snapshot
         of the current profile data.
         """
+        # Do profile data collection immediately on this unit
         if os.path.exists(COLLECT_PROFILE_DATA):
             subprocess.check_output([COLLECT_PROFILE_DATA])
 
+        # Tell the benchmark-gui charm the action_id via the benchmark
+        # relation. Benchmark-gui will pass the action_id to all
+        # collector charms in the environment (via the collector relation),
+        # triggering profile data collection on each.
+        if os.environ['CHARM_DIR']:
```

Comment:
Oh, duh, None is the default for `.get`. Also, Adam has a fair point about benchmark-start being invoked by someone silly. I suppose it probably should error out, but much earlier. Just check during `__init__()` for `CHARM_DIR` and one other variable (can't remember what it is, but it's like `JUJU_SOCKET` or something, which if present means most juju commands will work